File: code/my_fschool_backend/src/main/java/prm393/se1911/assignment/myfschoolbackend/model/response/StudentProfileResponse.java
package prm393.se1911.assignment.myfschoolbackend.model.response;

import lombok.Builder;

import java.time.LocalDate;

@Builder
public record StudentProfileResponse(
        String id,
        String studentCode,
        String fullName,
        LocalDate dateOfBirth,
        String gender,
        String avatarUrl,
        ClassResponse currentClass,
        AttendanceResponse todayAttendance
) {
}

File: code/my_fschool_frontend/lib/model/response/student_profile_response.dart
import 'package:dart_mappable/dart_mappable.dart';

import 'attendance_response.dart';
import 'class_response.dart';
import 'club_response.dart';

part 'student_profile_response.mapper.dart';

@MappableClass(generateMethods: GenerateMethods.decode)
class StudentProfileResponse with StudentProfileResponseMappable {
  String id;
  String studentCode;
  String fullName;
  DateTime dateOfBirth;
  String gender;
  String? avatarUrl;
  ClassResponse? currentClass;
  AttendanceResponse? todayAttendance;
  ClubResponse? joinedClub;

  StudentProfileResponse({
    required this.id,
    required this.studentCode,
    required this.fullName,
    required this.dateOfBirth,
    required this.gender,
    this.avatarUrl,
    this.currentClass,
    this.todayAttendance,
    this.joinedClub,
  });

  factory StudentProfileResponse.fromJson(String json) =>
      StudentProfileResponseMapper.fromJson(json);

  factory StudentProfileResponse.fromMap(Map<String, dynamic> map) =>
      StudentProfileResponseMapper.fromMap(map);
}
